src/generation.rs

```diff
@@ -216,12 +215,17 @@ impl<'a> DeserializeConfig<'a> {
     }
 }
 
-fn concat_files(paths: Vec<&str>) -> std::io::Result<String> {
+fn concat_files<P: AsRef<Path>>(paths: &Vec<P>) -> std::io::Result<String> {
     let mut buf = String::new();
     for path in paths {
         buf.push_str(
             &std::fs::read_to_string(path)
-                .map_err(|_| panic!("can't read: {}", path))
+                .map_err(|_| {
```

Comment:
@SimonIT  I just had a thought that since it was only happening in an error message that maybe `Path::to_string_lossy` would be good here instead of panicing. Oh well, the PR is already merged. If we ever try and remove some of the panics (since people are using it as a library now) we could do that here. In our usage as an application they're pretty equivalent and the panic makes things easier to code. I guess it depends how people using it as a library would handle it, if they would just have it fail as-is. It's probably not too useful to just migrate over some of the panics unless we do all of them, and doing it everywhere would require so many places to return results. However, they might mostly be contained to the parsing code and thus not be quite as bad. I suppose until then you could run it with one of the std::panic catching functions and then it doesn't matter much. Is that how you are handling it when using it as a library?